ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/TransactionReaper.java

```diff
@@ -931,6 +975,8 @@ public static synchronized long transactionLifetime()
     public static final long defaultCheckPeriod = 120000; // in milliseconds
     public static final long defaultCancelWaitPeriod = 500; // in milliseconds
     public static final long defaultCancelFailWaitPeriod = 500; // in milliseconds
+    public static final long defaultUntracedPeriod = 180000; // in milliseconds
+    public static final long defaultTracePeriod = 30000; // in milliseconds
```

Comment:
It is intended for use in prod, so it does need to be relatively cheap, but I think we're there. For ordinary use you'd set the gracePeriod greater than the default tx timeout, which means it would take no snapshots and have no overhead. When you want diagnostics, you temporarily lower it enough to catch the tx you are worried about. It's similar to support asking user to enable debug logging, but a much lighter weight, focussed alternative.

Two quibbles: Like most arj config properties, you have to restart to change it, which may not be desirable in prod environments. It could be read directly from the bean instead of cached in a reaper field, and thus be changeable through the mbean interface, but that may make it slower. Second, to turn it off entirely you have to set it not only higher than the default timeout, but higher than any timeout a tx may set explicitly, which takes a bit of thought unless you just go for e.g. max int.  However, that probably only matters for e.g. the wildfly integration code's default, not for the one we hardcode in core.

Reply:
Okay we will capture this logic in the docs.
With regards to reading the property directly from the bean; let's see what the support team say (which I'd think would be Sephen).

Reply:
I understand that you have already discussed the feature with Stephen so I marked this one as resolved.